lib/tbd/psi.rb

```diff
@@ -1096,7 +1105,13 @@ def populateTBDdads(model, dads)
       holes += populateTBDkids(model, properties[:skylights])
     end
 
-    face = model.get_face(wire, holes)
+    # Populate dad's face, yet only with hinged kids.
+    hinged = []
+    holes.each do |hole|
+      hinged << hole unless hole.attributes.has_key?(:unhinged)
+    end
+    face = model.get_face(wire, hinged)
```

Comment:
[Tubular Daylight Device](https://bigladdersoftware.com/epx/docs/9-6/input-output-reference/group-daylighting.html#daylightingdevicetubular) domes (subsurfaces) often free-float above a parent roof surface. Labelling such cases as _unhinged_ as none of their edges are shared with the parent. Solution:

1. get/generate parent _face_ only with subsurfaces sharing same plane equation (i.e. _hinged_)
2. subsequently populate (elsewhere in psi.rb) TBD with _unhinged_ subsurfaces.

Reply:
Are skylights unhinged too since they don't share an edge?  Or is this only for non-planar subsurfaces?  Really the same thing can happen for windows and doors too (although doors should share a bottom edge).

Reply:
It seems like something in here could be added to Topolys.  Right now, Topolys Faces can have holes but the holes are supposed to be planar.

Reply:
I edited my original post. True, @macumber: most subsurfaces won't share edges with parent surfaces. But subsurface edges _must_ (in the OpenStudio/EnergyPlus universe) fall within the same 3D plane of their parent surface (with the exception of TDD domes). My original post (and the recently-commented TBD source code, which I will soon correct) is indeed misleading. In summary, no issues with not sharing edges, or non-planar surfaces.